src/test/java/duke/ui/UiTest.java

```diff
@@ -0,0 +1,78 @@
+package duke.ui;
+
+import duke.task.Task;
+import duke.task.ToDo;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class UiTest {
+    private Ui ui;
+    private ByteArrayOutputStream outputStream;
+    private List<Task> tasks;
+
+    @BeforeEach
+    public void setUp() {
+        // Initialize the Ui instance and set up the output stream to capture prints
+        ui = new Ui();
+        outputStream = new ByteArrayOutputStream();
+        System.setOut(new PrintStream(outputStream));
+
+        // Reset the "database" (tasks list) before each test
+        tasks = new ArrayList<>();
+    }
+
+    @Test
+    public void testShowWelcome() {
+        ui.showWelcome();
+        String expectedOutput = "____________________________________________________________\n" +
+                " Hello! I'm Duke\n" +
+                " What can I do for you?\n" +
+                "____________________________________________________________\n";
+        assertEquals(expectedOutput, outputStream.toString());
+    }
+
+    @Test
+    public void testShowGoodbye() {
+        ui.showGoodbye();
+        String expectedOutput = "____________________________________________________________\n" +
+                " Bye. Hope to see you again soon!\n" +
+                "____________________________________________________________\n";
+        assertEquals(expectedOutput, outputStream.toString());
+    }
+
+    @Test
+    public void testShowTaskAdded() {
```

Comment:
Underscores may be used in test method names using the following three part format featureUnderTest_testScenario_expectedBehavior()